mio/0002: a non-blocking connect's EINPROGRESS is "in progress", not an error.
Upstream: https://github.com/wasix-org/mio/pull/4

--- a/src/sys/wasi/tcp.rs
+++ b/src/sys/wasi/tcp.rs
@@ -28,11 +28,14 @@
 
 pub(crate) fn connect(socket: &net::TcpStream, addr: SocketAddr) -> io::Result<()> {
     let addr = socket_addr(&addr);
-    unsafe {
+    match unsafe {
         wasi::sock_connect(
             socket.as_raw_fd() as u32,
             &addr
-        ).map_err(|errno| io::Error::from_raw_os_error(errno.raw() as i32))
+        )
+    } {
+        Err(errno) if errno.raw() as i32 != libc::EINPROGRESS => Err(io::Error::from_raw_os_error(errno.raw() as i32)),
+        _ => Ok(()),
     }
 }
 
